workers/social/lib/social/models/group/index.coffee

```diff
@@ -1552,65 +1552,142 @@ module.exports = class JGroup extends Module
 
-        (next) =>
-          JName.one { name: @slug }, (err, name) ->
-            removeHelper name, err, next
+          errors.push { label, err }  if err
+          next()
 
-        (next) =>
-          @fetchPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+        removeHelper = (model, err, next, label) ->
 
-        (next) =>
-          @fetchDefaultPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+          if err or not model
+            return kallback label, next, err
 
-        (next) =>
-          JInvitation = require '../invitation'
-          JInvitation.remove { groupName: @slug }, (err) ->
-            next err
+          model.remove (err) -> kallback label, next, err
 
-        (next) =>
-          ComputeProvider = require '../computeproviders/computeprovider'
-          ComputeProvider.destroyGroupResources this, -> next()
+        index = 0
 
-        (next) =>
-          JSession = require '../session'
-          @sendNotification 'GroupDestroyed', @slug, =>
-            JSession.remove { groupName: @slug }, (err) ->
-              next err
+        queue = [
 
-        (next) =>
-          @constructor.emit 'GroupDestroyed', this
-          next()
+          (next) ->
+            JApiToken = require '../apitoken'
+            JApiToken.remove { group: slug }, (err) ->
+              kallback 'JApiToken', next, err
+
+          (next) =>
+            ComputeProvider = require '../computeproviders/computeprovider'
+            ComputeProvider.destroyGroupResources this, -> next()
+
+          (next) ->
+            JGroupData = require './groupdata'
+            JGroupData.remove { slug }, (err) ->
+              kallback 'JGroupData', next, err
+
+          (next) ->
+            JInvitation = require '../invitation'
+            JInvitation.remove { groupName: slug }, (err) ->
+              kallback 'JInvitation', next, err
+
+          (next) =>
+            @fetchPermissionSet (err, permSet) ->
+              removeHelper permSet, err, next, 'PermissionSet'
+
+          (next) =>
+            @fetchDefaultPermissionSet (err, permSet) ->
+              removeHelper permSet, err, next, 'DefaultPermissionSet'
+
+          (next) ->
+            JName.one { name: slug }, (err, name) ->
+              removeHelper name, err, next, 'JName'
+
+          (next) =>
+            Relationship.remove { sourceId : @getId() }, next
+
+          (next) ->
+            url = '/api/social/payment/subscription/delete'
+            { deleteReq } = require '../socialapi/requests'
+
+            deleteReq url, { sessionToken }, (err, body) ->
+              err = null  if err?.description is 'not found'
+              kallback 'Subscription', next, err?.error
+
+          (next) =>
+            @remove (err) -> kallback 'JGroup', next, err
+
+          (next) =>
+            JSession = require '../session'
+            @sendNotification 'GroupDestroyed'
```

Comment:
we are sending "GroupDestroyed" event multiple times here on the retry case
same applies to line 1670